### PR TITLE
add section about geometry building to manual

### PR DESCRIPTION
**Pull request by tyrasd**

### CODE REVIEW

**Review by bonaparten — DISMISSED** [review]

File: documentation/manual/geometries.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Relations
+---------
+
+Relations are handeled by the OSHDB in two different ways: [Multipolygons](https://wiki.openstreetmap.org/wiki/Multipolygon) are converted to either [`Polygon`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Polygon.html) or [`MultiPolygon`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/MultiPolygon.html) geometries (depending on the number of outer rings), while all [other relation types](https://wiki.openstreetmap.org/wiki/Types_of_relation) result in a [`GeometryCollection`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/GeometryCollection.html).
```
Comment (joker234) [inline documentation/manual/geometries.md:21]:
```suggestion
Relations are handled by the OSHDB in two different ways: [Multipolygons](https://wiki.openstreetmap.org/wiki/Multipolygon) are converted to either [`Polygon`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Polygon.html) or [`MultiPolygon`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/MultiPolygon.html) geometries (depending on the number of outer rings), while all [other relation types](https://wiki.openstreetmap.org/wiki/Types_of_relation) result in a [`GeometryCollection`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/GeometryCollection.html).
```

Comment (joker234) [reply documentation/manual/geometries.md:21]: > …, while all [other relation types](https://wiki.openstreetmap.org/wiki/Types_of_relation) result in a [`GeometryCollection`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/GeometryCollection.html).

Is this true for Relations containing only one Geometry/Feature?

File: documentation/manual/geometries.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Invalid OSM Data
+----------------
+
+There are situations where a part of OSM's entities have either incomplete or invalid data, for example a broken multipolygon resulting from an mapping error. The OSHDB makes a best effort to return a (potentially [invalid](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#isValid--)) geometry also for these objects. For performance reasons the OSHDB cannot check for every possible error in the input (OSM) data, and for similar reasons it also cannot correct all errors it does find. This means that the OSHDB does not provide any guaranted outcome for all possible errors and might return an invalid or valid geometry as a result in such cases or even no result at all.
```
Comment (joker234) [inline documentation/manual/geometries.md:26]:
```suggestion
There are situations where a part of OSMs entities have either incomplete or invalid data, for example a broken multipolygon resulting from a mapping error. The OSHDB makes the best effort to return a (potentially [invalid](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#isValid--)) geometry also for these objects. For performance reasons the OSHDB cannot check for every possible error in the input (OSM) data, and for similar reasons it also cannot correct all errors it does find. This means that the OSHDB does not provide any guaranteed outcome for all possible errors and might return an invalid or valid geometry as a result in such cases or even no result at all. 
```

File: documentation/manual/geometries.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Invalid OSM Data
+----------------
+
+There are situations where a part of OSM's entities have either incomplete or invalid data, for example a broken multipolygon resulting from an mapping error. The OSHDB makes a best effort to return a (potentially [invalid](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#isValid--)) geometry also for these objects. For performance reasons the OSHDB cannot check for every possible error in the input (OSM) data, and for similar reasons it also cannot correct all errors it does find. This means that the OSHDB does not provide any guaranted outcome for all possible errors and might return an invalid or valid geometry as a result in such cases or even no result at all. 
+
+diff --git a/documentation/manual/README.md b/documentation/manual/README.md
```
Comment (joker234) [inline documentation/manual/geometries.md:28]:
```suggestion
```

**Review by joker234 — CHANGES_REQUESTED** [review]

File: documentation/manual/geometries.md (hunk 1 above)
Comment (tyrasd) [reply documentation/manual/geometries.md:21]: yes, I think so: https://github.com/GIScience/oshdb/blob/master/oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/geometry/OSHDBGeometryBuilder.java#L145

File: documentation/manual/geometries.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+OSM Feature Geometries
+======================
+
+The [_views_](views.md) provided by the OSHDB API provide direct access to the OSM entities, but also allows to get the [JTS](https://en.wikipedia.org/wiki/JTS_Topology_Suite#Geometry_model) [geometries](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html) of the OSM features corresponding to their state at the requested _snapshot_ or _contribution_ time(s). With this geometry, further operations such as [_filtering_](filters.md#areaOfInterest) or [_aggregation_](aggregation.md#aggregateByGeometry) can be performed. It is also irreplacable during the [_map-reduce_](map-reduce.md#geometry_helpers) step to generate useful results, such as the length of a road network for example.
```
Comment (bonaparten) [inline documentation/manual/geometries.md:4]: allow

File: documentation/manual/geometries.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+For some OSM elements, such as nodes, generating geometries is straight forward, for others the conversion requires further knowledge of the [data model](https://wiki.openstreetmap.org/wiki/Elements) and [tagging schema](https://wiki.openstreetmap.org/wiki/Tags) used by OSM. The following document gives an overview of how the OSHDB handles the building of geometries of different OSM object types.
+
+Nodes
+-----
+
+Nodes are always presented as [`Point`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Point.html)s. Note that nodes which have never had a tags and are a part of a way or relation are considered to be structural-only points (sometimes called _vertices_), and thus not returned in an OSHDB query when querying all nodes. This is because the OSHDB does not consider them to not be _map features_ by their own. They can however be fetched as members of their parent way or relation objects, if needed.
```
Comment (bonaparten) [inline documentation/manual/geometries.md:11]: a tag

Comment (bonaparten) [reply documentation/manual/geometries.md:11]: on their own

Comment (tyrasd) [reply documentation/manual/geometries.md]: > on their own

this one was already correct, wasn't it: _“…depending on their composition…”_?

**Review by joker234 — DISMISSED** [review]

**Review by joker234 — APPROVED** [review]

**Pull request merged**